### PR TITLE
Add categories to drink list

### PR DESCRIPTION
**Pull request by audiodude**
Currently the categories appear in the order that they were found in the dink list (all_drinks), which is probably a bug. We could hardcode the category order in DrinkService.

Here are some screenshots:

screenshot 2016-06-27 16 27 56
screenshot 2016-06-27 16 28 08

### CODE REVIEW

**Comment by audiodude** [comment]
This PR includes the Robo-thumbs PR btw.

**Comment by funrollloops** [comment]
Looks awesome! I'm tempted to deploy this tonight....but let's make sure
everything else works first.

File: static/components/drinks/drinks-service.ts [hunk 1]
```diff
@@ -46,9 +46,24 @@ class DrinksService {
...
         var pantry: Pantry = args[1];
-        resolve(recipe_response.data.filter((recipe) => {
+        var db = recipe_response.data.filter((recipe) => {
           return pantry.hasAllIngredients(recipe);
-        }));
+        });
+        var categories = {};
+        angular.forEach(db, (recipe) => {
+          angular.forEach(recipe.categories, (category) => {
+            var category_slug = this.slugifyDrinkName(category);
```
Comment (funrollloops) [inline static/components/drinks/drinks-service.ts]: We should probably rename the slugifyDrinkName method

File: static/drink-list/drink-list.html [hunk 2]
```diff
@@ -35,16 +36,21 @@
...
+                                 filter:{$:ctrl.state.query} |
+                                 orderBy:'drink_name'">
+          <img ng-src="thumbs/{{ctrl.slugify(drink.drink_name)}}.jpg" class="md-avatar">
+          <div class="md-list-item-text">
+            <h3>{{drink.drink_name}}</h3>
+            <p>{{ctrl.ingredientsCsv(drink)}}</p>
+          </div>
+        </md-list-item>
+      </div>
+        <md-list-item></md-list-item>
```
Comment (funrollloops) [inline static/drink-list/drink-list.html]: nit: indentation

**Comment by audiodude** [comment]
I've made some changes to make categories show in a specific order. Part of it is that I've stopped abusing slugifyDrinkName.

**Comment by funrollloops** [comment]
Nice!

**Pull request merged**